src/octtrees.rs

```diff
@@ -0,0 +1,90 @@
+use crate::bbox::BBox;
+use crate::buffers::Buffer;
+use crate::vector::{Vec3, Color};
+
+type Child<T> = Option<Box<T>>;
```

Comment:
This might have to be a `Rc` instead of `Box` to allow multiple pointers. Not sure about this though.